portal-impl/src/com/liferay/portal/service/impl/AddressLocalServiceImpl.java

```diff
@@ -249,6 +262,38 @@ public List<Address> getListTypeAddresses(
 			classPK, listTypeIds, start, end, orderByComparator);
 	}
 
+	@Override
+	public Address getOrAddIncompleteAddress(
+			String externalReferenceCode, long companyId, long userId,
```

Comment:
please keep just `String externalReferenceCode, long companyId, long userId, String className, long classPK`

Reply:
and jump the validation on the other fields checking if we are creating the incomplete address